synapse/storage/databases/main/receipts.py

```diff
@@ -122,21 +122,42 @@ async def get_users_with_read_receipts_in_room(self, room_id: str) -> Set[str]:
         receipts = await self.get_receipts_for_room(room_id, ReceiptTypes.READ)
         return {r["user_id"] for r in receipts}
 
-    @cached(num_args=2)
+    @cached()
     async def get_receipts_for_room(
         self, room_id: str, receipt_type: str
     ) -> List[Dict[str, Any]]:
+        """
+        Fetch the event IDs for the latest receipt for all users in a room with the given receipt type.
+
+        Args:
+            room_id: The room ID to fetch the receipt for.
+            receipt_type: The receipt type to fetch.
+
+        Returns:
+            A list of dictionaries of the user ID and event ID of the latest receipt for each user.
```

Comment:
I had to read the source to understand what was returned. I thought each dictionary was a map from multiple user IDs to event IDs. Perhaps:

> A list of dictionaries, one for each user ID. Each dictionary contains a user ID and the event ID of that user's latest receipt.

Reply:
This is also a suboptimal way of returning this data, but that's for a different PR...